### PR TITLE
Feature/unaligned instruction

### PR DESCRIPTION
**Pull request by nsurbay**
Regions in ExecBlockManager must be mutually exclusive.
This PR should fix this property when the code have instruction in instruction.

### CODE REVIEW

File: include/QBDI/Range.h [hunk 1]
```diff
@@ -34,29 +34,42 @@ template<typename T> class Range {
         return a > b ? a : b;
     }
 
+    T _start;  /*!< Range start value. */
```
Comment (romainthomas) [inline include/QBDI/Range.h:37]: Why moving to private ?

File: src/ExecBlock/ExecBlockManager.cpp [hunk 2]
```diff
@@ -216,50 +222,158 @@ size_t ExecBlockManager::searchRegion(rword address) const {
     LogDebug("ExecBlockManager::searchRegion", "Searching for address 0x%" PRIRWORD, address);
     // Binary search of the first region to look at
     while(low + 1 != high) {
-        size_t idx = (low + high) / 2;
-        if(regions[idx].covered.start > address) {
+        size_t idx = (low + high) >> 1;
```
Comment (romainthomas) [inline src/ExecBlock/ExecBlockManager.cpp:225]: This optimization is automatically done by the compiler, you should keep ``/ 2`` to express you intent

File: include/QBDI/Range.h (hunk 1 above)
Comment (nsurbay) [reply include/QBDI/Range.h:37]: `_start` is move to private to keep the consistency of the range. 

``` cpp
Range<rword> t {3, 5};
t._start = 20; // The range {20, 5} isn't valid.
```

With python, I may use `@property` and `@start.setter` but I don't find a c++ equivalent.

File: src/ExecBlock/ExecBlockManager.cpp (hunk 2 above)
Comment (nsurbay) [reply src/ExecBlock/ExecBlockManager.cpp]: ack

**Pull request merged**